Firestore/Source/Local/FSTLevelDBMutationQueue.mm

```diff
@@ -158,17 +162,23 @@ + (FSTBatchID)loadNextBatchIDFromDB:(std::shared_ptr<DB>)db {
   if (it->Valid() && [rowKey decodeKey:it->key()]) {
     moreUserIDs = YES;
     nextUserID = rowKey.userID;
+    batch_iter->Seek(tableKey);
+    it->Next();
   }
 
   // This loop assumes that nextUserId contains the next username at the start of the iteration.
   while (moreUserIDs) {
     // Compute the first key after the last mutation for nextUserID.
     auto userEnd = [FSTLevelDBMutationKey keyPrefixWithUserID:nextUserID];
     userEnd = util::PrefixSuccessor(userEnd);
+    Slice userEndSlice = userEnd;
 
     // Seek to that key with the intent of finding the boundary between nextUserID's mutations
-    // and the one after that (if any).
-    it->Seek(userEnd);
+    // and the one after that (if any). `batch_iter` trails by one entry, so it will point to
+    // the previous user's max batchID row.
+    for (; it->Valid() && it->key().compare(userEndSlice) < 0; it->Next()) {
```

Comment:
`it->Seek` isn't a linear search. This code was previously O(users * lg(mutations)) and is now O(mutations). We had an explicit goal for startup that we would not read the entire mutation queue. Apps that are offline for a while can end up with a significant startup delay if we read all mutations.

Note that this code runs at startup when we know no changes are in flight. I think this means we can just leave this as is and just not run it inside the new transaction.

Reply:
Oh yeah, that works too. I'll just skip the transaction for this step.